src/frontend/src/pages/FinancePage/ReimbursementRequestPage/ReimbursementRequestPage.tsx

```diff
@@ -0,0 +1,176 @@
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  const [height, setHeight] = useState<number>();
+
+  // doesnt work with the dependency array for some reason
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => {
+    if (ref.current) {
+      setHeight(ref.current.clientHeight + 350);
+    }
+  });
+
+  const theme = useTheme();
+
+  if (!reimbursementRequest) return <LoadingIndicator />;
+
+  const BasicInformationView = () => {
+    return (
+      <Box>
+        <Typography variant="h5">Basic Information</Typography>
+        <Grid container spacing={2} sx={{ ml: 2, mt: 2 }}>
+          <Grid item xs={12}>
+            <DetailDisplay label="Recipient" content={fullNamePipe(reimbursementRequest.recipient)} />
+          </Grid>
+          <Grid item xs={12}>
+            <DetailDisplay label="Purchased From" content={reimbursementRequest.vendor.name} />
+          </Grid>
+          <Grid item xs={12}>
+            <DetailDisplay label="Refund Source" content={`${reimbursementRequest.account}`} />
+          </Grid>
+          <Grid item xs={12}>
+            <DetailDisplay label="Date of Expense" content={`${datePipe(new Date(reimbursementRequest.dateOfExpense))}`} />
+          </Grid>
+          <Grid item xs={12}>
+            <DetailDisplay label="Expense Type" content={`${reimbursementRequest.expenseType.name}`} />
+          </Grid>
+        </Grid>
+      </Box>
+    );
+  };
+
+  const ReimbursementProductsView = () => {
+    const uniqueWbsElementsWithProducts = new Map<String, ReimbursementProduct[]>();
+    reimbursementRequest.reimbursementProducts.forEach((product) => {
+      const wbs = wbsPipe(product.wbsNum);
+      if (uniqueWbsElementsWithProducts.has(wbs)) {
+        const products = uniqueWbsElementsWithProducts.get(wbs);
+        products?.push(product);
+      } else {
+        uniqueWbsElementsWithProducts.set(wbs, [product]);
+      }
+    });
+
+    const keys = [];
+    for (const key of uniqueWbsElementsWithProducts.keys()) {
+      keys.push(key);
+    }
+
+    return (
+      <div ref={ref}>
+        <Typography variant="h5">Products</Typography>
+        <Table sx={{ mx: 2 }}>
+          <TableHead>
+            <TableRow>
+              <TableCell>WBS Element</TableCell>
+              <TableCell>Products</TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {keys.map((key) => {
+              return (
+                <TableRow>
+                  <TableCell>{key}</TableCell>
+                  <TableCell>
+                    {uniqueWbsElementsWithProducts.get(key)?.map((product) => {
+                      return <Chip label={`${product.name} $${product.cost}`} />;
+                    })}
+                  </TableCell>
+                </TableRow>
+              );
+            })}
+          </TableBody>
+        </Table>
+      </div>
+    );
+  };
+
+  const ReceiptsView = () => {
```

Comment:
I think these should be organized one receipt per row no matter what like it is in the mocks